router/handler_ctrl/updateCtrlAddresses.go

```diff
@@ -0,0 +1,50 @@
+package handler_ctrl
+
+import (
+	"github.com/michaelquigley/pfxlog"
+	"github.com/openziti/channel/v2"
+	"github.com/openziti/fabric/pb/ctrl_pb"
+	"google.golang.org/protobuf/proto"
+)
+
+var updateCtrlAddressesHandlerInstance *updateCtrlAddressesHandler
+
+type CtrlAddressUpdater interface {
+	UpdateCtrlEndpoints(endpoints []string) error
+}
+
+type updateCtrlAddressesHandler struct {
+	callback CtrlAddressUpdater
+
+	currentVersion uint64
+}
+
+func (handler *updateCtrlAddressesHandler) ContentType() int32 {
+	return int32(ctrl_pb.ContentType_UpdateCtrlAddressesType)
+}
+
+func (handler *updateCtrlAddressesHandler) HandleReceive(msg *channel.Message, ch channel.Channel) {
+	log := pfxlog.ContextLogger(ch.Label())
+	upd := &ctrl_pb.UpdateCtrlAddresses{}
+	if err := proto.Unmarshal(msg.Body, upd); err != nil {
+		log.WithError(err).Error("error unmarshalling")
+		return
+	}
+
+	if upd.IsLeader || handler.currentVersion == 0 || handler.currentVersion < upd.Index {
+		if err := handler.callback.UpdateCtrlEndpoints(upd.Addresses); err != nil {
+			log.WithError(err).Error("Unable to update ctrl addresses")
+		}
+		handler.currentVersion = upd.Index
```

Comment:
Maybe log the addresses here as well, unless they're already getting logged elsewhere?

Reply:
So it's getting logged in the `UpdateCtrlEndpoints`. Can add it here as a bulk message, in the aforementioned its logs them individually when they are being added/removed. Saying that out loud does tell me that already known endpoints are hidden so perhaps also log here as well actually?

Reply:
I think it's probably fine. As long as the change is logged somewhere not sure the entire set needs to be logged each time